pyk/src/pyk/ktool/kompile.py

```diff
@@ -28,10 +31,10 @@ def kompile(
 
     args = _build_arg_list(backend=backend, output_dir=output_dir, include_dirs=include_dirs, emit_json=emit_json)
 
-    proc_res = _kompile(str(main_file), *args)
-
-    if proc_res.returncode:
-        raise RuntimeError(f'Kompilation failed for: {main_file}')
+    try:
+        _kompile(str(main_file), *args)
+    except CalledProcessError as err:
+        raise RuntimeError(f'Command kompile exited with code {err.returncode} for: {main_file}', err.stdout, err.stderr)
 
     kompiled_dir = _kompiled_dir(main_file, output_dir)
```

Comment:
So I don't quite understand this, isn't the exception that `run_process` is throwing already enough? Or is this so we can have a more specfiic error message in this case?

Reply:
Good point. There are two reasons for doing it like this.

* Although `CalledProcessError` contains `stderr` and `stdout`, when the exception is uncaught, they will not be shown in the stack trace.
* We will eventually create some custom error class that we can raise instead of `RuntimeError`.